adapter/src/lib.rs

```diff
@@ -164,13 +164,13 @@ fn encode_params(
 
     let params: Vec<_> = types
```

Comment:
Probably this can be quite simplified and probably we can use `&str`, but it's not a job for this PR.